### DISCUSSION TITLE
Run on web cam

### DISCUSSION THREAD

**Discussion by krisgarces95** [SysCV/qd-3dt#9] [Q&A]
Hi, nice work. Can i run the software on web cam?

**Comment by eborboihuc** [comment]
The short answer is: yes, if you can extend the code!

Since our model has two inputs given, one is the image sequences, and the other is the camera extrinsics. You may need to estimate the [R|t] using visual odometry (off-the-shelf ones would be sufficient) on video input and build a data pipeline (image and camera matrix at least) stream to the model and do the prediction. If the camera is quite different from the one in the dataset, we also recommend fine-tuning the pre-trained weight for higher accuracy.

**Comment by BaophanN** [comment]
Is there any fork of pull request that has done this, thank you
